include/tune_quda.h

```diff
@@ -204,14 +210,17 @@ namespace quda {
     }
 
     /**
-     * @brief This can't be correctly queried in CUDA for all
-     * architectures so here we set set this.  Based on Table 14 of
-     * the CUDA Programming Guide 10.0 (Technical Specifications per
-     * Compute Capability).
+     * @brief Returns the maximum dynamic shared memory per block.
      * @return The maximum dynamic shared memory to CUDA thread block
      */
     unsigned int maxDynamicSharedBytesPerBlock() const
     {
+#if CUDA_VERSION >= 9000
+      static int max_shared_bytes = 0;
+      if (!max_shared_bytes) cudaDeviceGetAttribute(&max_shared_bytes, cudaDevAttrMaxSharedMemoryPerBlockOptin, comm_gpuid());
+      return max_shared_bytes;
+#else
+      // these variables are taken from Table 14 of the CUDA 10.2 prgramming guide
```

Comment:
do we need to have Volta hardcoded if we query it for CUDA 9+ ?

Reply:
Removed Volta